packages/react-native-codegen/src/parsers/utils.js

```diff
@@ -52,7 +52,50 @@ function createParserErrorCapturer(): [
   return [errors, guard];
 }
 
+function verifyPlatforms(
+  hasteModuleName: string,
+  moduleNames: string[],
+): $ReadOnly<{
+  cxxOnly: boolean,
+  excludedPlatforms: Array<'iOS' | 'android'>,
+}> {
+  let cxxOnly = false,
+    excludeIOS = false,
+    excludeAndroid = false;
+
+  const namesToValidate = [...moduleNames, hasteModuleName];
+
+  namesToValidate.forEach(name => {
+    if (name.endsWith('Android')) {
+      excludeIOS = true;
+    }
+
+    if (name.endsWith('IOS')) {
+      excludeAndroid = true;
+    }
+
+    if (name.endsWith('Cxx')) {
+      cxxOnly = true;
+      excludeIOS = true;
+      excludeAndroid = true;
+    }
```

Comment:
these are mutually exclusive, so we can return immediately, with something like:
```ts
if (name.endsWith) {
  //action
  return;
}
```

we can also move the `excludedPlatforms` before the `forEach` and remove the  `excludeIOS` and `excludeAndroid` booleans. 

something like:
```ts
let excludedPlatforms = [];
namesToValidate.forEach(name => {
    if (name.endsWith('Android')) {
      excludedPlatforms.push('iOS');
      return;
    }
    // ...
```

In this way, we can also get rid of the `if (excludeIOS) { }` and `if (excludeAndroid)` below.

Reply:
Thanks for the review!
The asked changes are done, I just used a `Set` instead of an array to avoid having duplicates in `excludedPlatforms`